### PR TITLE
chore(deps)!: update fake-timers to v14.0.0

### PR DESCRIPTION
**Pull request by xxzefgh**
### Description

Update `@sinonjs/fake-timers` to latest version and add support for `node:timers/promises` module.

 - Updated patches
 - Enabled new `ignoreMissingTimers` option by default as it better reflects previous behavior
 - Removed `setImmediate/clearImmediate` workaround as it has been fixed AFAIK
 - Updated existing tests to include all timers in global object as this behavior has changed in this commit (https://github.com/sinonjs/fake-timers/commit/341203310225bf5cd3d7396b2fcf276c5e218347)

I haven't added any new tests due to these checks making it difficult (not possible?):
[permalink](https://github.com/sinonjs/fake-timers/blob/48f089fdc830e39fcec31dd23099cc360da0bab2/src/fake-timers-src.js#L1848-L1853)
```ts
        if (_global === globalObject && timersModule) {
            clock.timersModuleMethods = [];
        }
        if (_global === globalObject && timersPromisesModule) {
            clock.timersPromisesModuleMethods = [];
        }
```

Partially fixes https://github.com/vitest-dev/vitest/issues/3088
closes https://github.com/vitest-dev/vitest/issues/7142

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by hi-ogawa — COMMENTED** [review]
Thanks for the PR :wave:
I just found a bug due to the old version 11.1.0 https://github.com/vitest-dev/vitest/issues/7142 and it makes to update this for Vitest 3.

> I haven't added any new tests due to these checks making it difficult (not possible?)

Do you mean it's hard to unit test in `test/core/test/fixtures/timers.suite.ts`? or `node:timers` and `node:timers/promise` fake don't even work with `vi.useFakeTimers`? (testing something locally, it looks like `globalThis.__vitest_required__` setup is too late?)

**Review by sheremet-va — DISMISSED** [review]

File: test/core/test/fixtures/timers.suite.ts [hunk 1]
```diff
@@ -24,28 +24,28 @@ describe('FakeTimers', () => {
 
   describe('construction', () => {
     it('installs setTimeout mock', () => {
-      const global = { Date: FakeDate, clearTimeout, process, setTimeout }
+      const global = { Date: FakeDate, clearTimeout, clearInterval, process, setTimeout, setInterval }
```
[the remark this answers is not in the document]
Comment (hi-ogawa) [reply test/core/test/fixtures/timers.suite.ts]: It looks like only these two are not technically necessary but others are required for tests. I just cleaned up a little https://github.com/vitest-dev/vitest/pull/7097/commits/e92778c4e16e3b38aa68a8ac9dcc1f516f1fc238

**Review by hi-ogawa — APPROVED** [review]

**Comment by xxzefgh** [comment]
@hi-ogawa sorry I couldn't respond, had a vacation. Thanks for your efforts!

**Comment by scamden** [comment]
Sorry to bug on here, but I see this was meant to enable timers/promises. I've upgraded to latest vite (3.1.3) and advanceTimersByTimeAsync doesn't seem to run my timers set with that module. Looking in the dist code I have installed under node_modules/vitest i don't see timers/promises anywhere inside the `requireFakeTimersSrc` function. Am I missing something here? Thanks in advance!

**Comment by hi-ogawa** [comment]
@scamden Please file an new issue with your reproduction so your question doesn't get lost. I'm not sure what exactly is your situation and what's expected.

**Comment by scamden** [comment]
> @scamden Please file an new issue with your reproduction so your question doesn't get lost. I'm not sure what exactly is your situation and what's expected.

fair enough! i just wanted to confirm that that this theoretically should work in that release now since the code didn't appear to be there (but i could certainly have missed something)

**Pull request merged**